packages/webapp/src/pages/Logs/components/TypesSelect.tsx

```diff
@@ -0,0 +1,118 @@
+import Button from '../../../components/ui/button/Button';
+import { useMemo, useState } from 'react';
+import { CrossCircledIcon } from '@radix-ui/react-icons';
+import { Popover, PopoverContent, PopoverTrigger } from '../../../components/ui/Popover';
+import { Command, CommandCheck, CommandEmpty, CommandGroup, CommandItem, CommandList } from '../../../components/ui/Command';
+import { typesOptions } from '../constants';
+import { Input } from '../../../components/ui/input/Input';
+
+export interface SearchableMultiSelectArgs<T> {
+    selected: string[];
+    onChange: (selected: T[]) => void;
+}
+
+export const TypesSelect: React.FC<SearchableMultiSelectArgs<any>> = ({ selected, onChange }) => {
+    const [open, setOpen] = useState(false);
+
+    const select = (val: string, checked: boolean) => {
+        if (val === 'all') {
+            onChange(['all']);
+            return;
+        }
+
+        let tmp = checked ? [...selected, val] : selected.filter((sel) => sel !== val);
+        const [type, action] = val.split(':');
+        if (!action && checked) {
+            // On check category, remove childs
+            tmp = tmp.filter((sel) => !sel.startsWith(`${type}:`));
+        } else if (action && checked) {
+            // On check child, remove parent
+            tmp = tmp.filter((sel) => sel !== type);
+        }
+        if (tmp.length > 1) {
+            tmp = tmp.filter((sel) => sel !== 'all');
+        }
+
+        onChange(tmp.length <= 0 ? ['all'] : tmp);
+    };
+
+    const reset = (e: any) => {
+        e.preventDefault();
+        onChange(['all']);
+    };
+
+    const isDirty = useMemo(() => {
+        return !(selected.length === 1 && selected[0] === 'all');
+    }, [selected]);
+
+    return (
+        <Popover open={open} onOpenChange={setOpen}>
+            <PopoverTrigger asChild>
+                <Button variant="zombieGray" size={'xs'}>
+                    Type
+                    {isDirty && (
+                        <button
+                            className="bg-pure-black text-white flex gap-1 items-center px-1.5 rounded-xl"
+                            onPointerDown={reset}
+                            onKeyDown={(e) => {
+                                if (['Enter', ' '].includes(e.key)) {
+                                    reset(e);
+                                }
+                            }}
+                        >
+                            <CrossCircledIcon />
+                            {selected.length}
```

Comment:
ux: The selected count makes the previous buttons jump and shift left and right when selecting. That's a bit strange I think

Reply:
yep agreed too, it's something I need to discuss with gabrielle